homeassistant/components/switch/xiaomi_miio.py

```diff
@@ -161,8 +220,69 @@ def async_update(self):
             self._state = state.is_on
             self._state_attrs.update({
                 ATTR_TEMPERATURE: state.temperature,
-                ATTR_LOAD_POWER: state.load_power,
+                ATTR_LOAD_POWER: state.load_power
             })
 
         except DeviceException as ex:
             _LOGGER.error("Got exception while fetching the state: %s", ex)
+
+
+class ChuangMiPlugV1Switch(XiaomiPlugGenericSwitch, SwitchDevice):
+    """Representation of a Chuang Mi Plug V1."""
+
+    def __init__(self, name, plug, device_info, channel_usb):
+        """Initialize the plug switch."""
+        name = name + ' USB' if channel_usb else name
+
+        XiaomiPlugGenericSwitch.__init__(self, name, plug, device_info)
+        self._channel_usb = channel_usb
+
+    @asyncio.coroutine
+    def async_turn_on(self, **kwargs):
+        """Turn a channel on."""
+        if self._channel_usb:
+            result = yield from self._try_command(
+                "Turning the plug on failed.", self._plug.usb_on)
+        else:
+            result = yield from self._try_command(
+                "Turning the plug on failed.", self._plug.on)
+
+        if result:
+            self._state = True
+            self._skip_update = True
+
+    @asyncio.coroutine
+    def async_turn_off(self, **kwargs):
+        """Turn a channel off."""
+        if self._channel_usb:
+            result = yield from self._try_command(
+                "Turning the plug on failed.", self._plug.usb_off)
+        else:
+            result = yield from self._try_command(
+                "Turning the plug on failed.", self._plug.off)
+
+        if result:
+            self._state = False
+            self._skip_update = True
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Fetch state from the device."""
+        from miio import DeviceException
+
+        # On state change the device doesn't provide the new state immediately.
+        if self._skip_update:
+            self._skip_update = False
+            return
```

Comment:
A question here, would it be that bad if the device provides an incorrect state for one run? I think it would make more sense and make the device adhere to how other devices (afaik) do it. So instead of trying to trick the system, just believing what comes out from the device.

Reply:
Yes, it's bad. If you turn on the switch at the user interface it will flip back immediately. 30 seconds later the correct state will be received. The user experience suffers without the workaround.